src/main/java/edu/kis/vh/nursery/FIFORhymer.java

```diff
@@ -1,30 +1,30 @@
 package edu.kis.vh.nursery;
 
-import edu.kis.vh.nursery.dataStructures.IntArrayStack;
+import edu.kis.vh.nursery.dataStructures.IntLinkedList;
 
 public class FIFORhymer extends DefaultCountingOutRhymer {
 
-    private final DefaultCountingOutRhymer temp = new DefaultCountingOutRhymer();
+    private final IntLinkedList temp = new IntLinkedList();
 
-    public FIFORhymer(IntArrayStack stack) {
+    public FIFORhymer(IntLinkedList stack) {
```

Comment:
IntLinkedList does not allow using IntArrayStack. Use IntStructure! Then apply appropriate constructors to ListRhymersFactory and ArrayRhymersFactory (point 2.3.1)